cuvis_ai/utils/serializer.py

```diff
@@ -31,115 +32,137 @@ class CuvisYamlLoader(yaml.SafeLoader):
-    return dumper.represent_scalar('!numpy.float32', str(float(data)))
+def numpy_float32_representer(
+    dumper: yaml.SafeDumper, data: np.float32
+) -> yaml.ScalarNode:
+    return dumper.represent_scalar("!numpy.float32", str(float(data)))
 
 
-def numpy_float32_constructor(loader, node):
+def numpy_float32_constructor(
+    loader: yaml.SafeLoader, node: yaml.ScalarNode
+) -> np.float32:
     value = loader.construct_scalar(node)
     return np.float32(value)
 
 
-def numpy_float64_representer(dumper, data):
-    return dumper.represent_scalar('!numpy.float64', str(float(data)))
+def numpy_float64_representer(
+    dumper: yaml.SafeDumper, data: np.float64
+) -> yaml.ScalarNode:
+    return dumper.represent_scalar("!numpy.float64", str(float(data)))
 
 
-def numpy_float64_constructor(loader, node):
+def numpy_float64_constructor(
+    loader: yaml.SafeLoader, node: yaml.ScalarNode
+) -> np.float64:
     value = loader.construct_scalar(node)
     return np.float64(value)
 
 
 CuvisYamlDumper.add_representer(np.ndarray, numpy_array_file_representer)
 CuvisYamlDumper.add_representer(np.float32, numpy_float32_representer)
 CuvisYamlDumper.add_representer(np.float64, numpy_float64_representer)
-CuvisYamlLoader.add_constructor(
-    '!numpy.ndarray.str', numpy_arrray_string_constructor)
-CuvisYamlLoader.add_constructor(
-    '!numpy.ndarray.file', numpy_arrray_file_constructor)
-CuvisYamlLoader.add_constructor('!numpy.float32', numpy_float32_constructor)
-CuvisYamlLoader.add_constructor('!numpy.float64', numpy_float64_constructor)
+CuvisYamlLoader.add_constructor("!numpy.ndarray.str", numpy_arrray_string_constructor)
+CuvisYamlLoader.add_constructor("!numpy.ndarray.file", numpy_arrray_file_constructor)
+CuvisYamlLoader.add_constructor("!numpy.float32", numpy_float32_constructor)
+CuvisYamlLoader.add_constructor("!numpy.float64", numpy_float64_constructor)
 
 
 class YamlSerializer(Serializer):
-
-    def __init__(self, data_dir: Path, filename: str = 'main') -> None:
+    def __init__(self, data_dir: Path, filename: str = "main") -> None:
         super().__init__(data_dir)
 
         self.filename = filename
 
     def serialize(self, data: dict) -> None:
-        with change_working_dir(self.data_dir):
-            with open(f'{self.filename}.yml', 'w') as f:
-                yaml.dump(data, f, Dumper=CuvisYamlDumper,
-                          default_flow_style=False)
+        with change_working_dir(self.data_dir), open(f"{self.filename}.yml", "w") as f:
+            yaml.dump(data, f, Dumper=CuvisYamlDumper, default_flow_style=False)
 
     def load(self) -> dict:
-        with change_working_dir(self.data_dir):
-            with open(f'{self.filename}.yml') as f:
-                data = yaml.load(f, Loader=CuvisYamlLoader)
+        # with change_working_dir(self.data_dir), open(f"{self.filename}.yml") as f:
+        # data = yaml.safe_load(f)  # Use safe_load instead of yaml.load
+        with open(f"{self.filename}.yml") as f:
```

Comment:
```suggestion
    with change_working_dir(self.data_dir), open(f"{self.filename}.yml") as f:
```
Still need to change to the correct working directory